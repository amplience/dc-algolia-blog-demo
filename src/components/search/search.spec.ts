import { mount } from '@vue/test-utils';
import Search from './search.vue';
import searchResponse from './fixtures/search-response.json';
import searchResponseMutli from './fixtures/search-response-multiple-pages.json';
import flushPromises from 'flush-promises';
import { MultiResponse, Response } from '~/node_modules/@types/algoliasearch';

const mockSearchFn = jest.fn();

jest.mock(
  '@/services/algolia/search-client',
  () =>
    function() {
      return {
        client: {
          search: () => mockSearchFn.apply(null, arguments)
        }
      };
    }
);

describe('Search', () => {
  afterAll(
    (): void => {
      jest.restoreAllMocks();
    }
  );

  test('is a search result', async () => {
    mockSearchFn.mockImplementationOnce(
      async (): Promise<MultiResponse<Response>> => {
        return { results: [searchResponse] };
      }
    );
    const wrapper = mount(Search);
    await flushPromises();
    expect(mockSearchFn).toBeCalled();
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });

  xtest('is a search result with pagination', async () => {
    mockSearchFn.mockImplementationOnce(
      async (): Promise<MultiResponse<Response>> => {
        return { results: [searchResponseMutli] };
      }
    );
    const wrapper = mount(Search);
    await flushPromises();
    const secondPage = wrapper.findAll('.ais-Pagination-link').at(1);
    secondPage.trigger('click');
    expect(mockSearchFn).toBeCalled();
    expect(wrapper.isVueInstance()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });

});
