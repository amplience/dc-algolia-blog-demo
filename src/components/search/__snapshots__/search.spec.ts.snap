// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search is a search result 1`] = `
<div
  class="ais-InstantSearch ais-InstantSearch--ssr"
>
  <ais-configure
    hitsperpage="9"
  />
   
  <div
    class="ais-SearchBox search"
  >
    <form
      action=""
      class="ais-SearchBox-form"
      novalidate="novalidate"
      role="search"
    >
      <input
        aria-label="Search"
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="ais-SearchBox-input"
        maxlength="512"
        placeholder="Search here…"
        required="required"
        spellcheck="false"
        type="search"
      />
       
      <button
        class="ais-SearchBox-submit"
        title="Search"
        type="submit"
      >
        <svg
          class="ais-SearchBox-submitIcon"
          height="10"
          role="img"
          viewBox="0 0 40 40"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <button
        class="ais-SearchBox-reset"
        hidden="hidden"
        title="Clear"
        type="reset"
      >
        <svg
          class="ais-SearchBox-resetIcon"
          height="1em"
          role="img"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <!---->
    </form>
  </div>
   
  <div
    class="ais-Hits"
  >
    <el-row
      gutter="20"
    >
      <el-col
        span="8"
      >
        <el-card
          body-style="[object Object]"
          shadow="hover"
        >
          <div>
            <nuxt-link
              to="/blogs/d79b4424-7fd4-4237-85e7-3df51c8716e4"
            >
              <img
                alt="Snippet Blog Post"
                class="image"
                src="//i1-qa.adis.ws/i/devedition01/woman-purple-dress"
              />
               
              <div
                class="content"
              >
                <h2>
                  Snippet Blog Post
                </h2>
                 
                <div
                  class="description"
                >
                  No description.
                </div>
              </div>
            </nuxt-link>
          </div>
        </el-card>
      </el-col>
    </el-row>
  </div>
   
  <div
    class="ais-Pagination"
  >
    <ul
      class="ais-Pagination-list"
    >
      <li
        class="ais-Pagination-item ais-Pagination-item--firstPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="First"
          class="ais-Pagination-link"
        >
          ‹‹
        </span>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--previousPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Previous"
          class="ais-Pagination-link"
        >
          ‹
        </span>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--selected"
      >
        <a
          class="ais-Pagination-link"
          href="#"
        >
          1
        </a>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--nextPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Next"
          class="ais-Pagination-link"
        >
          ›
        </span>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--lastPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Last"
          class="ais-Pagination-link"
        >
          ››
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Search is a search result with pagination 1`] = `
<div
  class="ais-InstantSearch ais-InstantSearch--ssr"
>
  <ais-configure
    hitsperpage="2"
  />
   
  <div
    class="ais-SearchBox search"
  >
    <form
      action=""
      class="ais-SearchBox-form"
      novalidate="novalidate"
      role="search"
    >
      <input
        aria-label="Search"
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="ais-SearchBox-input"
        maxlength="512"
        placeholder="Search here…"
        required="required"
        spellcheck="false"
        type="search"
      />
       
      <button
        class="ais-SearchBox-submit"
        title="Search"
        type="submit"
      >
        <svg
          class="ais-SearchBox-submitIcon"
          height="10"
          role="img"
          viewBox="0 0 40 40"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <button
        class="ais-SearchBox-reset"
        hidden="hidden"
        title="Clear"
        type="reset"
      >
        <svg
          class="ais-SearchBox-resetIcon"
          height="1em"
          role="img"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <!---->
    </form>
  </div>
   
  <div
    class="ais-Hits"
  >
    <el-row
      gutter="20"
    >
      <el-col
        span="8"
      >
        <el-card
          body-style="[object Object]"
          shadow="hover"
        >
          <div>
            <nuxt-link
              to="/blogs/d79b4424-7fd4-4237-85e7-3df51c8716e4"
            >
              <img
                alt="Blog Post"
                class="image"
                src="//i1-qa.adis.ws/i/devedition01/woman-purple-dress"
              />
               
              <div
                class="content"
              >
                <h2>
                  Blog Post
                </h2>
                 
                <div
                  class="description"
                >
                  No description.
                </div>
              </div>
            </nuxt-link>
          </div>
        </el-card>
      </el-col>
      <el-col
        span="8"
      >
        <el-card
          body-style="[object Object]"
          shadow="hover"
        >
          <div>
            <nuxt-link
              to="/blogs/d79b4424-7fd4-4237-85e7-3df51c8716e4"
            >
              <img
                alt="Blog Post"
                class="image"
                src="//i1-qa.adis.ws/i/devedition01/woman-purple-dress"
              />
               
              <div
                class="content"
              >
                <h2>
                  Blog Post
                </h2>
                 
                <div
                  class="description"
                >
                  No description.
                </div>
              </div>
            </nuxt-link>
          </div>
        </el-card>
      </el-col>
    </el-row>
  </div>
   
  <div
    class="ais-Pagination"
  >
    <ul
      class="ais-Pagination-list"
    >
      <li
        class="ais-Pagination-item ais-Pagination-item--firstPage"
      >
        <a
          aria-label="First"
          class="ais-Pagination-link"
          href="#"
        >
          ‹‹
        </a>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--previousPage"
      >
        <a
          aria-label="Previous"
          class="ais-Pagination-link"
          href="#"
        >
          ‹
        </a>
      </li>
       
      <li
        class="ais-Pagination-item"
      >
        <a
          class="ais-Pagination-link"
          href="#"
        >
          1
        </a>
      </li>
      <li
        class="ais-Pagination-item ais-Pagination-item--selected"
      >
        <a
          class="ais-Pagination-link"
          href="#"
        >
          2
        </a>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--nextPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Next"
          class="ais-Pagination-link"
        >
          ›
        </span>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--lastPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Last"
          class="ais-Pagination-link"
        >
          ››
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Search is passed the default configured number of search results 1`] = `
<div
  class="ais-InstantSearch ais-InstantSearch--ssr"
>
  <ais-configure
    hitsperpage="9"
  />
   
  <div
    class="ais-SearchBox search"
  >
    <form
      action=""
      class="ais-SearchBox-form"
      novalidate="novalidate"
      role="search"
    >
      <input
        aria-label="Search"
        autocapitalize="off"
        autocomplete="off"
        autocorrect="off"
        class="ais-SearchBox-input"
        maxlength="512"
        placeholder="Search here…"
        required="required"
        spellcheck="false"
        type="search"
      />
       
      <button
        class="ais-SearchBox-submit"
        title="Search"
        type="submit"
      >
        <svg
          class="ais-SearchBox-submitIcon"
          height="10"
          role="img"
          viewBox="0 0 40 40"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <button
        class="ais-SearchBox-reset"
        hidden="hidden"
        title="Clear"
        type="reset"
      >
        <svg
          class="ais-SearchBox-resetIcon"
          height="1em"
          role="img"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"
            fillRule="evenodd"
          />
        </svg>
      </button>
       
      <!---->
    </form>
  </div>
   
  <div
    class="ais-Hits"
  >
    <el-row
      gutter="20"
    >
      <el-col
        span="8"
      >
        <el-card
          body-style="[object Object]"
          shadow="hover"
        >
          <div>
            <nuxt-link
              to="/blogs/d79b4424-7fd4-4237-85e7-3df51c8716e4"
            >
              <img
                alt="Blog Post"
                class="image"
                src="//i1-qa.adis.ws/i/devedition01/woman-purple-dress"
              />
               
              <div
                class="content"
              >
                <h2>
                  Blog Post
                </h2>
                 
                <div
                  class="description"
                >
                  No description.
                </div>
              </div>
            </nuxt-link>
          </div>
        </el-card>
      </el-col>
      <el-col
        span="8"
      >
        <el-card
          body-style="[object Object]"
          shadow="hover"
        >
          <div>
            <nuxt-link
              to="/blogs/d79b4424-7fd4-4237-85e7-3df51c8716e4"
            >
              <img
                alt="Blog Post"
                class="image"
                src="//i1-qa.adis.ws/i/devedition01/woman-purple-dress"
              />
               
              <div
                class="content"
              >
                <h2>
                  Blog Post
                </h2>
                 
                <div
                  class="description"
                >
                  No description.
                </div>
              </div>
            </nuxt-link>
          </div>
        </el-card>
      </el-col>
    </el-row>
  </div>
   
  <div
    class="ais-Pagination"
  >
    <ul
      class="ais-Pagination-list"
    >
      <li
        class="ais-Pagination-item ais-Pagination-item--firstPage"
      >
        <a
          aria-label="First"
          class="ais-Pagination-link"
          href="#"
        >
          ‹‹
        </a>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--previousPage"
      >
        <a
          aria-label="Previous"
          class="ais-Pagination-link"
          href="#"
        >
          ‹
        </a>
      </li>
       
      <li
        class="ais-Pagination-item"
      >
        <a
          class="ais-Pagination-link"
          href="#"
        >
          1
        </a>
      </li>
      <li
        class="ais-Pagination-item ais-Pagination-item--selected"
      >
        <a
          class="ais-Pagination-link"
          href="#"
        >
          2
        </a>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--nextPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Next"
          class="ais-Pagination-link"
        >
          ›
        </span>
      </li>
       
      <li
        class="ais-Pagination-item ais-Pagination-item--lastPage ais-Pagination-item--disabled"
      >
        <span
          aria-label="Last"
          class="ais-Pagination-link"
        >
          ››
        </span>
      </li>
    </ul>
  </div>
</div>
`;
